src/components/HomePage/SearchResultsArea/Offer/OfferContent.js

```diff
@@ -28,75 +95,261 @@ const OfferContent = ({ offer, loading }) => {
                             {getRandomOngoingSearchMessage()}
                         </Typography>
                     </React.Fragment>
-
                 </div>
             </div>
         );
     } else {
         return (
             <div className={classes.offerContent} data-testid="offer-content">
-                {offer === null ?
+                <AdminReasonModal
+                    open={showAdminReasonModal}
+                    setOpen={setShowAdminReasonModal}
+                    offer={offer}
+                    setVisibilityState={setVisibilityState}
+                    visibilityState={visibilityState}
+                    dealWithPromiseError={dealWithPromiseError}
+                />
+                {(offer === null && !isPage) ?
                     <div className={classes.unselectedOffer} id="no_selected_offer_text">
                         <Typography variant="h5" classes={{ root: classes.pleaseSelectOfferText }}>
                             Please select an offer to view the details
                         </Typography>
                     </div>
                     :
-                    <React.Fragment>
-                        <div className={classes.offerHeader}>
-                            <Typography variant="h4" gutterBottom>
-                                {offer.title}
-                            </Typography>
+                    <div>
```

Comment:
This component has gotten too big I think, please subdivide the tree into smaller components (maybe an OfferDetails, and an OfferMessage - actually it should be OfferContent, but this is already OfferContent, unless you come up with a better name... 👀 )